lib/ConnectionManager.js

```diff
@@ -62,6 +62,16 @@ function ConnectionManager(connectionString, options, stateListener) {
     this.updateTimeout(options.sessionTimeout);
     this.connectTimeoutHandler = null;
 
+    /**
+     * add sessionTimeoutHandler
+     *
+     * @type {number}
+     */
+    this.sessionTimeoutHandler = setTimeout(
```

Comment:
You need to set the timer in `connect` rather than in the constructor. If the caller waits before calling `connect` you would fire too early.